chips/nrf52/src/chip.rs

```diff
@@ -26,7 +24,142 @@ impl<I: InterruptService> NRF52<I> {
     }
 }
 
-impl<I: InterruptService> kernel::Chip for NRF52<I> {
+/// This macro defines a struct that, when initialized,
+/// instantiates all peripheral drivers shared across all supported nrf52 chips
+/// in Tock. If a board
+/// wishes to use only a subset of these peripherals, this
+/// macro cannot be used, and this struct should be
+/// constructed manually in main.rs. The input to the macro is the name of the struct
+/// that will hold the peripherals, which can be chosen by the board.
+#[macro_export]
+macro_rules! create_default_nrf52_peripherals {
+    ($N:ident) => {
```

Comment:
Why is this name a parameter? I only see the macro invoked with "Nrf52BasePeripherals".

Reply:
It doesn't really have to be a parameter. I made it a parameter so that if someone wanted to switch between the macro version and a version with fewer drivers instantiated in boards, it would be a little easier. But I am fine with a version that does not pass a name.